From: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Date: Tue, 10 Oct 2006 13:13:55 +0000 (+0900)
Subject: [MIPS] Do not use -msym32 option for modules.
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=c3a0eb881338608c32ff01da824da63e84aa6c8b;hp=ac5c84216a96351e2083f45e86295d934856873c

[MIPS] Do not use -msym32 option for modules.

On 64-bit kernel, modules are loaded into XKSEG for now.  While XKSEG
address is not a sign-extended 32-bit address, we can not use -msym32
option.

Signed-off-by: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from ca78b1a5c6a6e70e052d3ea253828e49b5d07c8a commit)
---

diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index 7b3a4fc..6aba9d4 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -63,7 +63,9 @@ cflags-y		+= -mabi=64
 ifdef CONFIG_BUILD_ELF64
 cflags-y		+= $(call cc-option,-mno-explicit-relocs)
 else
-cflags-y		+= $(call cc-option,-msym32)
+# -msym32 can not be used for modules since they are loaded into XKSEG
+CFLAGS_MODULE		+= $(call cc-option,-mno-explicit-relocs)
+CFLAGS_KERNEL		+= $(call cc-option,-msym32)
 endif
 endif
 
